.github/workflows/lean4checker_nightly.yml

```diff
@@ -0,0 +1,128 @@
+name: lean4checker on nighly-testing-YYYY-MM-DD Workflow
+
+## This workflow is identical to the `lean4checker.yml` workflow, except that it runs on the
+## most recent `nightly-testing-YYYY-MM-DD` tag instead of the `master` branch.
+## Please keep the workflow up to date with `lean4checker.yml`.
+
+on:
+  schedule:
+    - cron: '0 0 * * *'   # Runs at 00:00 UTC every day
+  workflow_dispatch:
+
+jobs:
+  check-lean4checker:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Fetch latest tags
+        run: |
+          git fetch --tags
+          # Find the most recent tag matching the pattern
+          LATEST_TAG=$(git tag | grep -E '^nightly-testing-[0-9]{4}-[0-9]{2}-[0-9]{2}$' | sort -r | head -n 1)
+          echo "LATEST_TAG=\"${LATEST_TAG}\"" >> "$GITHUB_ENV"
+
+      - name: Checkout latest tag branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.LATEST_TAG }}
```

Comment:
```suggestion
```
I think this will be somewhat easier to maintain if we preserve the order of corresponding steps between the two files (not sure if I've done it properly here).